auditlog: add exports/imports

ijon opened the pull request:
Add audit logging for (database/tables) export and import operations.

Cancel/Forget requests for export/imports are still not audit logged.

KIKIMR-21797

### Changelog entry

Add audit logging for (database/tables) export and import operations.

### Changelog category

* Improvement

ijon committed:
auditlog: add exports/imports

Add audit logging for (database/tables) export and import operations.

Export/import cancel/forget requests are still not logged.

KIKIMR-21797

ijon commented on ydb/core/tx/schemeshard/schemeshard_build_index__create.cpp:
```diff
@@ -240,17 +241,8 @@ class TSchemeShard::TIndexBuilder::TTxCreate: public TSchemeShard::TIndexBuilder
-
-        return it->second;
+        return true;
     }
```
`GetUid` has moved to schemeshard_xxport__helpers.*

ijon commented on ydb/core/tx/schemeshard/ut_login/ut_login.cpp:
```diff
@@ -80,12 +61,6 @@ Y_UNIT_TEST_SUITE(TSchemeShardLoginTest) {
         UNIT_ASSERT(describe.GetPathDescription().GetDomainDescription().GetSecurityState().PublicKeysSize() > 0);
     }
 
-    NAudit::TAuditLogBackends CreateTestAuditLogBackends(std::vector<std::string>& buffer) {
-        NAudit::TAuditLogBackends logBackends;
-        logBackends[NKikimrConfig::TAuditConfig::TXT].emplace_back(new TMemoryLogBackend(buffer));
-        return logBackends;
-    }
-
```
`CreateTestAuditLogBackends` moved to `ut_helpers/auditlog_helpers.*`

molotkov-and commented on ydb/core/tx/schemeshard/schemeshard_audit_log.cpp:
```diff
@@ -167,6 +178,209 @@ void AuditLogModifySchemeTransactionDeprecated(const NKikimrScheme::TEvModifySch
+
+        AUDIT_PART("paths", RenderList(record.Paths), !record.Paths.empty())
+
+        AUDIT_PART("export_uid", record.Uid);
```
This is common function for import and export but only `export_uid` is used in this line. Perhaps need `xport_uid`

molotkov-and commented on ydb/core/tx/schemeshard/schemeshard_audit_log.cpp:
```diff
@@ -167,6 +178,209 @@ void AuditLogModifySchemeTransactionDeprecated(const NKikimrScheme::TEvModifySch
+
+        // AUDIT_PART("start_time", exportInfo.StartTime.ToString(), exportInfo.StartTime != TInstant::Zero())
+        // AUDIT_PART("end_time", exportInfo.EndTime.ToString(), exportInfo.EndTime != TInstant::Zero())
+        AUDIT_PART("start_time", record.StartTime)
+        AUDIT_PART("end_time", record.EndTime)
```
Do we need to check `record.StartTime` and `record.EndTime` for emptiness and set default stub? These values don`t set in `*xportStart` function

ijon replied:
It is intentional. Start record shouldn't have times at all (it can have `start_time`, but why, and cannot have `end_time`). And end record must have both times.
And that is responsibility of `_AuditLogXxport{Start,End}()`

molotkov-and commented on ydb/core/tx/schemeshard/ut_restore/ut_restore.cpp:
```diff
@@ -49,6 +50,18 @@ namespace {
         Aws::ShutdownAPI(Options);
     }
 
+    std::string FindAuditLine(const std::vector<std::string>& auditLines, const std::string& substr) {
```
There is same function in `ut_export.cpp`. May be move it to `ut_helpers/auditlog_helpers.*`?

ijon replied:
Certainly. Thank you for spotting this.

ijon committed:
fixup

- rename `tx_id` to `id`
- rename `export_uid` to `uid`
- add testing for `uid`
- add additional parts for export to yt

ijon committed:
fixup: move FindAuditLine() to auditlog_helpers
